packages/studio/src/utils/PageDataValidator.ts

```diff
@@ -0,0 +1,87 @@
+export interface ValidationResult {
+  valid: boolean;
+  errorMessages: string;
+}
+
+/**
+ * PageDataValidator contains various static utility methods
+ * for validation of user-inputted page data.
+ */
+export default class PageDataValidator {
+  private isEntityPage = false;
+
+  constructor(isEntityPage?: boolean) {
+    if (isEntityPage) {
+      this.isEntityPage = isEntityPage;
+    }
+  }
+
+  checkIsURLEditable(url?: string) {
+    if (!url) {
+      return false;
+    }
+    const result: ValidationResult = this.validate({ url });
+    return result.valid;
+  }
+  /**
+   * Throws an error if the user-inputted page data is invalid.
+   */
+  validate(pageData: { pageName?: string; url?: string }): ValidationResult {
+    const errorMessages: string[] = [];
+    if (pageData.pageName !== undefined)
+      errorMessages.push(...this.validatePageName(pageData.pageName));
+    if (pageData.url !== undefined)
+      errorMessages.push(
+        ...this.validateURLSlug(pageData.url, this.isEntityPage)
+      );
+    return {
+      valid: errorMessages.length === 0,
+      errorMessages: errorMessages.join("\r\n"),
+    } as ValidationResult;
+  }
+
+  /**
+   * Throws an error if the page name is invalid.
+   */
+  private validatePageName(pageName: string) {
+    const errorMessages: string[] = [];
+    if (!pageName) {
+      errorMessages.push("A page name is required.");
+    }
+    const errorChars = pageName.match(/[\\/?%*:|"<>]/g);
```

Comment:
This will still allow these characters through. Do we want to filter more of them out?

[]{}()!@#$^&_+-=,.;'`~😂